### DISCUSSION TITLE
Feature request: Drag&Drop Jupyter Notebooks in issues

### DISCUSSION THREAD

**Discussion by kolibril13** [community/community#50144] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

Hi!

It would be amazing if there was a way to simply share notebooks in GitHub issues via Drag & Drop.

### What's the problem?

Currently, reporting python issues that occur in notebooks is time-consuming.  It requires multiple copy+paste steps and manual code-block formatting. Furthermore, the notebook format is lost, so error messages won't have red highlights, images won't show the same size, cell numbering is lost, and copy+paste mistakes can happen.

### So why notebooks in issues?

It lowers the barrier to writing a python related issues by reducing the involved mental routine work: embedding structured notebooks would become as simple as dropping a video or an image to an issue. 

The process of issue writing can mostly happen in the notebook using markdown cells. Iterating on a bug report becomes easier because code-re-execution is possible while phrasing the issue, which makes it easier to stay in the creative flow.

Maintainers can directly see the nicely formatted notebook content including code-blocks, error messages, images and cell numbers.

Maintainers can simply download the notebook and reproduce the error without multiple copy+pasting steps.

### Possible Alternatives

Screenshots/Screencasts of notebooks, but these are inconvenient for maintainers.
Notebook upload to Gist or GitHub, but that's a lot of manual work, and the notebook won't appear on the GitHub issue page itself.

**Comment by maartenbreddels** [comment]
Totally agree. I do that as well, explore an issue in a notebook, and then start copying code and copy-pasting screenshots into an issue. Being able to drag in a notebook would be much better.

**Comment by manzt** [comment]
Dropping here in case it is useful for anyone. I rediscovered an alias I had set for helping me preview notebooks as markdown in the terminal. I guess it could also come in handy for this case, by piping the contents to the clipboard:

```hs
alias nbcat="jupyter nbconvert --log-level WARN --stdout --to markdown $1"
```

```sh
nbcat Untitled.ipynb | pbcopy
```

https://github.com/community/community/assets/24403730/bd79089a-3fea-4b58-944b-23b128a5b4b5
